### ISSUE TITLE
Kafka datasource kill task waits indefinitely 

### ISSUE THREAD

**Issue by iercan** [apache/druid#10981]
### Affected Version

Druid version: 0.20.1

### Description

I've created a kafka supervisor which create ingestion task per hour. When I create a kill task it stays wait state even if supervisor change to new task

In the picture supervisor creates task at 8:35. I finished at 9:35 and I'd expect kill task take place but it still can not start and new ingestion task start. As a result kill task wait forever. How can I overcome this? 

![image](https://user-images.githubusercontent.com/3406152/110772437-aa4a3180-826c-11eb-8e04-b54b9d81f03a.png)

I see below log in overlord. Looks like it is a priority issue but couldn't figure out how to resolve.
```
2021-03-11T08:01:13,810 INFO [TaskQueue-Manager] org.apache.druid.indexing.overlord.TaskLockbox - Cannot create a new taskLockPosse for request..... taskIds=[....]}]] have same or higher priorities
```

**Comment by FrankChen021** [comment]
It seems it's the same problem as #10508. One solution is to cancel the kill task and issue a kill task manually through HTTP interface by limiting the interval parameter.

**Comment by iercan** [comment]
@FrankChen021 How to set interval? I'd appreciated if you can explain in details

**Comment by FrankChen021** [comment]
Look at [this](https://druid.apache.org/docs/latest/tutorials/tutorial-delete-data.html#run-a-kill-task)

The request body looks like 
```
{
  "type": "kill",
  "dataSource": "your-data-source-name",
  "interval" : "2015-09-12/2015-09-13"
}
```

The range of interval should not include the intervals of running ingestion tasks, or the kill task would wait ingestion tasks to complete.

**Comment by iercan** [comment]
@FrankChen021 I resolved issue by increasing start delay of tasks managed by supervisor. So that kill task can have a time to start 

![image](https://user-images.githubusercontent.com/3406152/110793408-a37be880-8285-11eb-9e6b-f437b6480bc4.png)

**Comment by remizyaka** [comment]
> @FrankChen021 I resolved issue by increasing start delay of tasks managed by supervisor. So that kill task can have a time to start
> 
> ![image](https://user-images.githubusercontent.com/3406152/110793408-a37be880-8285-11eb-9e6b-f437b6480bc4.png)

For some reason it doesn't help me. this delay only works on first task run, any other consequent ingestion kafka tasks will run immediately after previous finished or even before it finished. So if I have task duration set to 1 hour and startDelay set to PT300S and start supervisor at 16:17, first task will start at 16:22 and second will start at 17:22 -  no delay before starting.
Maybe I did it in a wrong place? ioConfig->startDelay of kafka supervisor  payload.
